### ISSUE TITLE
New 'role-focused Configurations' do not match functionality of existing Configurations

### ISSUE THREAD

**Issue by aSemy** [gradle/gradle#27266]
### Current Behavior

I get an unhelpful error when I try to migrate to the new 'role-focused Configurations'.

> Dependencies can not be declared against the `testNew` configuration

The 'role-focused Configurations' are [not documented](https://docs.gradle.org/8.5/userguide/declaring_dependencies.html#sec:defining-custom-configurations).

### Expected Behavior

I do not get an unhelpful error when I try to migrate to the new 'role-focused Configurations'.

The 'role-focused Configurations' ~are~ [not documented](https://docs.gradle.org/8.5/userguide/declaring_dependencies.html#sec:defining-custom-configurations).

### Steps to Reproduce

Currently I define a Configuration, [following the docs](https://docs.gradle.org/8.5/userguide/declaring_dependencies.html#sec:defining-custom-configurations):

```kotlin
val testCurrent = configurations.create("testCurrent") {
  isCanBeResolved = true
  isCanBeConsumed = false
}

dependencies {
  testCurrent("x:y:123")
}
```

When I try to upgrade it to the new docs, I get an unhelpful error.

```kotlin
val testCurrent = configurations.resolvable("testCurrent").get()

dependencies {
  testCurrent("x:y:123")
}
```

> Dependencies can not be declared against the `testCurrent` configuration

This error doesn't make sense to me, and doesn't help me find the correct solution.

### Gradle version

8.5

**Comment by aSemy** [comment]
> @aSemy BTW you need two configurations now, one dependencyScope containing your dependencies and another resolvable one which is able to resolve the dependencies and provides the actual jars (or xyz).

Thanks, that wasn't clear from the documentation. It would be helpful if there were some good docs for this because to me it looks unnecessary and just makes Gradle more complicated with no benefit. I still only want and need one Configuration for downloading some files, but now Gradle requires I make two?

**Comment by hfhbd** [comment]
> I still only want and need one Configuration for downloading some files, but now Gradle requires I make two?

Yes, it is even documented a few sections up: https://docs.gradle.org/8.5/userguide/declaring_dependencies.html#sec:resolvable-consumable-configs
I don't know the real reason, but reusing a configuration for different scopes is deprecated and will be result into an error starting Gradle 9, afaik.

**Comment by hfhbd** [comment]
And the Jasper example should also be improved to reflect the "new" roles and to not use one configuration for dependencyScope as well as resolving the classpath.

**Comment by aSemy** [comment]
> Yes, it is even documented a few sections up: https://docs.gradle.org/8.5/userguide/declaring_dependencies.html#sec:resolvable-consumable-configs I don't know the real reason, but reusing a configuration for different scopes is deprecated and will be result into an error starting Gradle 9, afaik.

Oh, I mean now Gradle wants to make it so two Configurations are required for downloading, whereas now only one is required.

```kotlin

// currently only one Configuration is needed for resolving files
val myResolver by configurations.registering {
    isCanBeResolved = true
    isCanBeConsumed = false
}


// but now Gradle wants to require an additional one, which just seems to increase the amount of boilerplate code


val myResolver2 = configurations.dependencyScope("myResolver2") {

}

val myResolver2Boilerplate = configurations.resolvable("myResolver2Boilerplate") {
    extendsFrom(myResolver.get())
}
```

Fortunately Gradle seems to still only needs a single 'Provider' Configuration.

If 3 Configurations are really required, can't Gradle just wrap them up into a single 'DependencyContainer' class?
